functions/go/starlark/README.md

```diff
@@ -33,9 +42,69 @@ source: |
   run(ctx.resource_list["items"], "prod")
 ```
 
-`StarlarkRun` has the following field beside the standard KRM fields:
-- `source`: (required) The source code of the Starlark script.
+`StarlarkRun` has the following field besides the standard KRM fields:
+
+- `source`: (required) A multi-line string that contains the source code of the
+  Starlark script.
+
+### Developing Starlark Script
+
+In Starlark, a [for loop] is permitted only within a function definition. It
+means if you want to iterate over `ctx.resource_list["items"]`, it has to be in
+a function. You can refer the example `functionConfig` above.
+
+Here's what you can do in the Starlark script:
+
+- Read resources from `ctx.resource_list`. The `ctx.resource_list` complies with
+  the [KRM Functions Specification]. You can read the input resources from
+  `ctx.resource_list[items]` and the `functionConfig` from
+  `ctx.resource_list[functionConfig]`.
+- Write resources to `ctx.resource_list[items]`.
+- Return an error using [`fail`][fail].
+- Write error message to stderr using [`print`][print]
+
+Here's what you cannot do in the Starlark script:
+
+- We don't support the `results` field, i.e. if you write some results in
```

Comment:
We should support it at some point? If so, it's a current limitation and you want to use "yet" here.

Reply:
Done